src/Drupal/Commands/core/DeployHookCommands.php

```diff
@@ -0,0 +1,185 @@
+                        'hook' => $hook,
+                        'description' => $description,
+                    ];
+                }
+            }
+        }
+
+        return new RowsOfFields($rows);
+    }
+
+    /**
+     * Run pending deploy update hooks.
+     *
+     * @usage deploy:hook
+     *   Run pending deploy hooks.
+     *
+     * @command deploy:hook
+     */
+    public function run()
+    {
+        $pending = $this->registry->getPendingUpdateFunctions();
+
+        if (empty($pending)) {
+            $this->logger()->success(dt('No pending deploy hooks.'));
+            return self::EXIT_SUCCESS;
+        }
+
+        $process = $this->processManager()->drush($this->siteAliasManager()->getSelf(), 'deploy:hook-info');
+        $process->mustRun();
+        $this->output()->writeln($process->getOutput());
+
+        if (!$this->io()->confirm(dt('Do you wish to run the specified pending deploy hooks?'))) {
+            throw new UserAbortException();
+        }
+
+        $success = true;
+        if (!$this->getConfig()->simulate()) {
+            try {
+                foreach ($pending as $function) {
+                    $func = new \ReflectionFunction($function);
+                    $this->logger()->notice('Deploy hook started: ' . $func->getName());
+
+                    // Pretend it is a batch operation to keep the same signature
+                    // as the post update hooks.
+                    $sandbox = [];
+                    do {
+                        $return = $function($sandbox);
+                        if (!empty($return)) {
+                            $this->logger()->notice($return);
+                        }
+                    } while (isset($sandbox['#finished']) && $sandbox['#finished'] < 1);
+
+                    $this->registry->registerInvokedUpdates([$function]);
+                    $this->logger()->debug('Performed: ' . $func->getName());
+                }
+            } catch (\Throwable $e) {
+                $variables = Error::decodeException($e);
+                unset($variables['backtrace']);
+                $this->logger()->error(dt('%type: @message in %function (line %line of %file).', $variables));
+                $success = false;
+            }
+        }
+
+        $level = $success ? ConsoleLogLevel::SUCCESS : LogLevel::ERROR;
+        $this->logger()->log($level, dt('Finished performing deploy hooks.'));
+        return $success ? self::EXIT_SUCCESS : self::EXIT_FAILURE;
+    }
+
+    /**
+     * Marks a deploy update hook as not having run.
+     *
+     * During development one often wants to re-run the hook, so this helps
+     * re-setting it so that it can be run again.
+     *
+     * @usage deploy:hook-reset mymodule_deploy_runagain
+     *   Unregisters that a deploy hook has run so that it runs again.
+     *
+     * @param string $hook
+     *   The hook name to reset.
+     *
+     * @command deploy:hook-reset
```

Comment:
How hard would it to provide a reset for hook_update_n and post_update? I think we should do them all or do none.

Reply:
It would not be hard, the code for post_update hooks would look almost the same (with a different key-value store) the hook_update_n would look like `drupal_set_installed_schema_version("woot", 8103)`.

We can also skip it.. I just found it to be convenient to use while working on the deploy hook, it would indeed also be convenient for the other hooks. So maybe we remove it here and then add it in a second step for all?

Reply:
I removed it for now. we can add a new PR later